src/audio/google/google_rtc_audio_processing_mock.c

```diff
@@ -145,11 +147,28 @@ int GoogleRtcAudioProcessingProcessCapture_int16(GoogleRtcAudioProcessingState *
 	int16_t *ref = state->aec_reference;
 	int16_t *mic = (int16_t *) src;
 	int16_t *out = dest;
-	int n;
-
+	int n, io, im, ir;
+
+	/* Mix input and reference channels to output. If the number of input
+	 * channels and reference channels is equal or higher than output then
+	 * matching channels are mixed. If output channels count is higher than
+	 * input or reference, the mic or reference channels indices to mix are
+	 * restarted from zero. This is nothing like real AEC operation but a
+	 * scheme to mix to every channel for testing.
+	 */
 	memset(dest, 0, sizeof(int16_t) * state->num_output_channels * state->num_frames);
 	for (n = 0; n < state->num_frames; ++n) {
-		*out = *mic + *ref;
+		im = 0;
+		ir = 0;
+		for (io = 0; io < state->num_output_channels; io++) {
+			out[io] = sat_int16((int32_t)mic[im++] + ref[ir++]);
+			if (im == state->num_capture_channels)
```

Comment:
You might mix several time same channel like is output is 2 and input is one channel you'll get 2 times the first channel.
```
 out[io] = sat_int16(
            (im < state->num_capture_channels ? (int32_t)mic[im++] : 0) + 
            (ir < state->num_reference_channles ? ref[ir++] : 0)
            );
```
Might be more appropriate (sorry for the vey poor codestyle).

Reply:
We don't really know what's best for testing with mock-up. Sure, I can update it to be like this.

Reply:
@lkoenig Thanks, I included your suggestion as such, it works well!